send_message_msteams_and_aws_script.py

```diff
@@ -0,0 +1,277 @@
+    QUOTABLE_API_URL = "https://api.quotable.io/random"
+
+    def get_content(self):
+        response = requests.get(self.QUOTABLE_API_URL)
+
+        if response.status_code != 200:
+            logging.error("Failed to retrieve a quote")
+            return None
+        else:
+            data = response.json()
+            quote_text = data["content"]
+            quote_author = data["author"]
+            logging.info("A quote successfully generated.")
+
+            return quote_text, quote_author
+
+
+class ImageGeneratorFromPiscumAPI(ContentGenerator):
+    WIDTH = random.randint(300, 800)
+    HEIGHT = random.randint(300, 800)
+    PISCUM_API_URL = "https://picsum.photos"
+
+    def get_content(self):
+        image_url = f"{self.PISCUM_API_URL}/{self.WIDTH}/{self.HEIGHT}"
+        return image_url
+
+
+class TeamsMessageSender:
+    MESSAGE_COLOR = "#F8C471"
+
+    def __init__(self, webhook_url):
+        self.webhook_url = webhook_url
+
+    def send_message(self, title, quote_text, quote_author, image_url):
+        """Method combines the whole message and send it to MSTeams channel."""
+        myTeamsMessage = pymsteams.connectorcard(self.webhook_url)
+        myTeamsMessage.color(self.MESSAGE_COLOR)
+        myTeamsMessage.title(title)
+        myTeamsMessage.text(
+            f"**Quote:** {quote_text}\n\n"
+            f"**Author:** {quote_author}\n\n"
+            f"![Image]({image_url})"
+        )
+
+        try:
+            myTeamsMessage.send()
+            logging.info("Message successfully sent.")
+            return
+        except (ConnectionError, HTTPError, TimeoutError) as e:
+            logging.error(
+                f"Failed to send message due to network error: {e}",
+                exc_info=True
+            )
+        except pymsteams.TeamsWebhookException as e:
+            logging.error(
+                f"Failed to send message due to Teams webhook error: {e}",
+                exc_info=True
+            )
+        except ValueError as e:
+            logging.error(
+                f"Failed to send message due to value error: {e}",
+                exc_info=True
+            )
+
+
+class S3Loader:
+    def __init__(self):
+        self.s3_key = None
+        self.s3_client = boto3.client(
+            service_name='s3',
+            aws_access_key_id='test',
+            aws_secret_access_key='test',
+            endpoint_url='http://localhost:4566',
+        )
+
+        self.bucket_name = "images-msteams"
+
+    def extract_image_name(self, url):
+        parsed_url = urlparse(url)
+        img_name = parsed_url.path.split("/")[-1]
+        return img_name
+
+    def image_upload(self, image_url):
+        try:
+            response = requests.get(image_url)
+            response.raise_for_status()
+            image_data = response.content
+
+            with BytesIO() as f:
+                f.write(image_data)
+
+            timestamp = datetime.now().strftime("%Y%m%d%H%M%S")
```

Comment:
It might be worthy to pass a time from airflow tasks and format according to logical execution time, not the current timestamp

Reply:
done